Documentation/Sandboxing.md

```diff
@@ -0,0 +1,14 @@
+## SerenityOS Sandboxing
+
+SerenityOS provides pledge and unveil syscalls to achieve opt-in sandboxes for its applications. However for user software like image editors or PDF viewers we want users to have the choice of opening new files and adding them to the sandbox as needed.
+
+### FileSystemAccessServer
+
+Designed to be used after unveiling, the FileSystemAccessServer is an opt-in service which allows users to choose whether additional files on their system can be read from or written to.
+
+The promise provided here is that only files and directories included in unveil calls or approved by a user action should be allowed to be read.
+
+User actions that could indicate approval include:
```

Comment:
Just thinking out loud, is there any way an attacker could automate these interactions today so an action we think would have to be authorized by a user, could be taken without the user actually being involved? Via IPC to windows server or some other mechanism? I believe this kind of vulnerability is possible on windows for example, and leads to all kinds of unfortunate security issues.